substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/JavaMainWrapper.java

```diff
@@ -133,6 +137,10 @@ public static int runCore(int paramArgc, CCharPointerPointer paramArgv) {
         int exitCode;
         try {
             if (SubstrateOptions.ParseRuntimeOptions.getValue()) {
+                if (SubstrateOptions.InstallExitHandlers.getValue()) {
+                    Target_java_lang_Terminator.setup();
+                }
```

Comment:
Maybe you should register a startup hook instead.

Reply:
I agree: ec15571605e2cf13390133e908dea82b7884654e